connect/src/com/telenor/mobileconnect/MobileConnectSdkProfile.java

```diff
@@ -112,23 +115,74 @@ public Uri getAuthorizeUri(Map<String, String> parameters, List<String> locales)
     }
 
     @Override
-    public DoNext onStartAuthorization(Map<String, String> parameters) {
-        OperatorDiscoveryAPI.OperatorDiscoveryResult odResult = null;
+    public void onStartAuthorization(
+            final Map<String, String> parameters,
+            final OnStartAuthorizationCallback callback) {
 
-        final String msisdn = readPhoneNumber();
-        if (msisdn != null) {
-            odResult = getOperatorDiscoveryResult(msisdn);
-        }
+        final Callback<OperatorDiscoveryAPI.OperatorDiscoveryResult> odCallbackForMccMnc =
+                new Callback<OperatorDiscoveryAPI.OperatorDiscoveryResult>() {
+                    @Override
+                    public void success(OperatorDiscoveryAPI.OperatorDiscoveryResult operatorDiscoveryResult, Response response) {
+                        if (isInitialized || initialize(operatorDiscoveryResult)) {
```

Comment:
Also you actually care only about WellKnown stuff here, so maybe it'd be better if you check for it, rather than checking for initialization (and hide wellknown behind it).